src/jit/scopeinfo.cpp

```diff
@@ -154,6 +154,80 @@ CodeGenInterface::siVarLoc::siVarLoc(const LclVarDsc* varDsc, regNumber baseReg,
     }
 }
 
+//------------------------------------------------------------------------
+// Equals: Compares first reference and then values of the structures.
+//
+// Arguments:
+//    lhs   - a "siVarLoc *" to compare.
+//    rhs   - a "siVarLoc *" to compare.
+//
+// Notes:
+//    Return true if both are nullptr.
+bool CodeGenInterface::siVarLoc::Equals(const siVarLoc* lhs, const siVarLoc* rhs)
+{
+    // are both nullptr or the same reference
+    bool areEquals = lhs == rhs;
+
+    if (!areEquals && nullptr != lhs && nullptr != rhs && lhs->vlType == rhs->vlType)
+    {
+        // If neither is nullptr, and are not the same reference, compare values
+        switch (lhs->vlType)
+        {
+            case VLT_STK:
+            case VLT_STK_BYREF:
+                areEquals = lhs->vlStk.vlsBaseReg == rhs->vlStk.vlsBaseReg;
```

Comment:
That would be much better with C++20 default comparisons where you can ask for default `<=>` and get all this code go away.

Reply:
I didn't know its existence, thanks for sharing!

Reply:
It says the compiler generates the code for comparing structs. I wonder how it deals with Union?